cli/remotebazel/remotebazel.go

```diff
@@ -501,10 +499,6 @@ func Run(ctx context.Context, opts RunOpts, repoConfig *RepoConfig) (int, error)
 
 	ctx = metadata.AppendToOutgoingContext(ctx, "x-buildbuddy-api-key", opts.APIKey)
 
-	instanceHash := sha256.New()
-	instanceHash.Write(uuid.NodeID())
-	instanceHash.Write([]byte(repoConfig.Root))
```

Comment:
Should we keep this around for local machines only, so that users won't get assigned to other users' snapshots unexpectedly? On GitHub, the env var `CI=true` will be set, so maybe if CI=true then we can set `instanceHash = ""` ?

Reply:
it seems like it might also be useful to keep `repoConfig.Root` around as the instance hash, regardless of whether CI=true or not

Reply:
Yeah I was going back and forth if it would be more beneficial or annoying to use others' snapshots, but I think it would be more beneficial in the longterm. It would result in faster build times for everyone. It would be like a global analysis cache. If you want to use a user-specific snapshot, you can always set something like --runner_exec_properties=user=maggie

I logged `repoConfig.Root` and it includes the full directory path. I think that the repo path on github runners can change. [Here](https://github.com/buildbuddy-io/buildbuddy/blob/master/enterprise/server/hostedrunner/hostedrunner.go#L157)we default the affinity key to the URL, which seems sufficient to me

Reply:
If you're running `bb remote` for the first time, then it will be nice to be matched up to a warm analysis cache, but when running `bb remote` multiple times in succession (i.e. interactively), if you have a lot of users doing that, then Bazel will constantly be switching Git branches and there will be lots of cache evictions. I think that's what the original intent of the "session" key was - we'd have per-user sessions that would not get clobbered by other users' sessions, so when you are running `bb` iteratively, you keep getting matched up the same VM and it feels like you're just developing locally.

Reply:
We do include the branch name in the snapshot key, so this feels like a reasonable trade off to me? I'd assume most users will be on a branch

The dream is that remote bazel would be even better than developing locally. It would be awesome if you always hit a warm bazel instance, even if your local machine would not have

Reply:
I have the same workflow as Brandon (working off master). Doesn't hurt to experiment though.

Reply:
Oh interesting. An idea for the future is we could see if a snapshot exists for a user session and only if it doesn't do we use a more general snapshot. Similar idea to what we do with fallback github branch keys now